gooresmed/download.py

```diff
@@ -124,6 +124,41 @@ class Download(_DownloadBase):
            ``start`` to the end of the media.
     """
 
+    def _prepare_request(self):
+        """Prepare the contents of an HTTP request.
+
+        .. _sans-I/O: https://sans-io.readthedocs.io/
+
+        This is everything that must be done before a request that doesn't
+        require network I/O (or other I/O). This is based on the `sans-I/O`_
+        philosophy.
+
+        Returns:
+            dict: The headers for the request.
+
+        Raises:
+            ValueError: If the current :class:`Download` has already
+                finished.
+        """
+        if self.finished:
+            raise ValueError('A download can only be used once.')
+
+        headers = {}
+        _add_bytes_range(self.start, self.end, headers)
+        return headers
```

Comment:
@jonparrott I'm a little "torn" here. I consider just returned all the inputs for the request:

```
pos_args = (self.media_url,)
kw_args = {'headers': headers}
return pos_args, kw_args
```